CMakeLists.txt

```diff
@@ -112,6 +112,13 @@ if(CMAKE_SYSTEM_NAME STREQUAL "WASI")
         "SHELL:$<$<COMPILE_LANGUAGE:Swift>:-Xcc -D_WASI_EMULATED_MMAN>")
 endif()
 
+# Enable whole module optimization for release builds & incremental for debug builds
+if(POLICY CMP0157)
+    set(CMAKE_Swift_COMPILATION_MODE "$<IF:$<CONFIG:Release>,wholemodule,incremental>")
```

Comment:
I would recommend putting this almost directly under the `project` call that enables Swift so that someone doesn't accidentally add a target before the compilation mode is set.